cli/azd/pkg/azapi/stack_deployments.go

```diff
@@ -285,7 +285,11 @@ func (d *StackDeployments) DeployToSubscription(
 
 	_, err = poller.PollUntilDone(ctx, nil)
 	if err != nil {
-		return nil, err
+		deploymentError := createDeploymentError(err)
```

Comment:
I'm wondering if we would want to do something similar for DeployToResourceGroup.

Reply:
yes!
added it